dashboard/src/shared/schema.ts

```diff
@@ -1,17 +1,110 @@
+  if (splittedPath.length === 1) {
+    // If the path doesn't contain inner objects, simply return
+    return rootElem.get(path);
+  }
+  // If the path contains inner elements, iterate until finding the last one
+  let res = rootElem.get(splittedPath[0]);
+  splittedPath.slice(1, splittedPath.length).forEach(p => (res = res && res.get(p)));
+  return res;
 }
 
+// setValue modifies the current values (text) based on a path
 export function setValue(values: string, path: string, newValue: any) {
-  // TBD
-  return values;
+  const doc = YAML.parseDocument(values);
+  if (path.includes(".")) {
+    // If the path points to an inner element we need to parse it
+    // and obtain the previous element to set it
+    // e.g.
+    // path = "a.b.c"
+    // itemToSet = "c"
+    // elem = "a.b"
+    // elem.set(itemToSet, value)
+    const splittedPath = path.split(".");
+    const itemToSet = splittedPath[splittedPath.length - 1];
+    splittedPath.pop();
+    const pathToSet = splittedPath.join(".");
+    const elem = getDocumentElem(doc, pathToSet);
+    if (!elem) {
+      // TODO: It should be possible to handle this situation and add a new object
+      throw new Error(`Unable to set the value ${path}. Parent element not found`);
+    }
+    elem.set(itemToSet, newValue);
+  } else {
+    doc.set(path, newValue);
+  }
+  return doc.toString();
+}
+
+// getDefaultValues returns the current value of an object based on YAML text and its path
+export function getDefaultValue(defaultValues: string, path: string): string | undefined {
+  const doc = YAML.parseDocument(defaultValues);
+  if (path.includes(".")) {
+    const elem = getDocumentElem(doc, path);
+    // If the path is found, return it as a string
+    return elem && elem.toString();
+  }
+  return path.includes(".") ? getDocumentElem(doc, path).toString() : doc.get(path);
+}
+
+// lookForFormParams iterates recursively over JSON Schema properties and returns IBasicFormParams
+function lookForFormParams(
+  defaultValues: string,
+  schemaProperties: {},
+  parentPath?: string,
+  params?: IBasicFormParam[],
+) {
+  Object.keys(schemaProperties).map(propertyKey => {
+    // The param path is its parent path + the object key
+    const itemPath = `${parentPath || ""}${propertyKey}`;
+    // If the property has the key "form", it's a basic parameter
+    if (schemaProperties[propertyKey].form) {
+      const newParam = {
+        // The name of the param is identified by the form key
+        name: schemaProperties[propertyKey].form,
+        path: itemPath,
+        // Use the default value either from the JSON schema or the default values
+        value: getDefaultValue(defaultValues, itemPath) || schemaProperties[propertyKey].default,
+      };
+      params = params ? params.concat(newParam) : [newParam];
+    }
+    // If the property is an object, iterate recursively
+    if (schemaProperties[propertyKey].type === "object") {
+      params = lookForFormParams(
```

Comment:
`params.concat(lookForFormParams(...))` recursively here (but without passing in the `params` arg, obviously).

Or why do you need to rely on both a return value and an array reference as a function arg?

Reply:
took me a while to get my head around that one :sweat_smile: 

You are right, the `params` argument is not needed.